### ISSUE TITLE
WindowServer: Error on boot: Rejecting path '/tmp/portal/inspectables' since it hasn't been unveiled with 'r' permission.

### ISSUE THREAD

**Issue by x-zvf** [SerenityOS/serenity#7108]
I've just checked out and built Serenity from master.
The following error is printed in the console (during the GUI boot):
```
[#0 WindowServer(13:13)]: Rejecting path '/tmp/portal/inspectables' since it hasn't been unveiled with 'r' permission.
```
The error in the boot context:
```
[#0 mount(8:8)]: mount tmp @ /tmp
[#0 mount(8:8)]: VFS: Mounting TmpFS at /tmp (inode: 1:6049) with flags 5
[#0 mount(8:8)]: mount: successfully mounted /tmp
SystemServer(6:6): Creating /tmp/coredump directory
SystemServer(6:6): Read command line: Kernel/Kernel hello
SystemServer(6:6): Booting in graphical mode
SystemServer(6:6): Core::EventLoop: Failed to connect to InspectorServer
SystemServer(6:6): Activating 24 services...
AudioServer(11:11): Started thread "AudioServer[mixer]", tid = 24
LookupServer(9:9): Using network config file at /etc/LookupServer.ini
[#0 WindowServer(13:13)]: Rejecting path '/tmp/portal/inspectables' since it hasn't been unveiled with 'r' permission.
[#0 WindowServer(13:13)]: 0xc02ca9a3  Kernel::VFS::validate_path_against_process_veil(AK::StringView, int) +0x309
[#0 WindowServer(13:13)]: 0xc02cf056  Kernel::VFS::resolve_path(AK::StringView, Kernel::Custody&, AK::RefPtr<Kernel::Custody, AK::RefPtrTraits<Kernel::Custody> >*, int, int) +0x200
[#0 WindowServer(13:13)]: 0xc02cfd64  Kernel::VFS::open(AK::StringView, int, unsigned short, Kernel::Custody&, AK::Optional<Kernel::UidAndGid>) +0x18a
[#0 WindowServer(13:13)]: 0xc031e944  Kernel::LocalSocket::connect(Kernel::FileDescription&, AK::Userspace<sockaddr const*>, unsigned int, Kernel::ShouldBlock) +0x302
[#0 WindowServer(13:13)]: 0xc0486512  Kernel::Process::sys$connect(int, AK::Userspace<sockaddr const*>, unsigned int) +0x94e
[#0 WindowServer(13:13)]: 0xc03be6d6  syscall_handler +0x1e7d
[#0 WindowServer(13:13)]: 0xc03bc821  syscall_asm_entry +0x31
```

The GUI and Inspector seem to be working fine, so I am not sure, what, if any effect this error has on the functionality of  the system.
But I figure that having a kernel stack-trace during boot is probably not expected behavior.

**Comment by x-zvf** [comment]
The error originates from `Userland/Libraries/LibCore/EventLoop.cpp` on line 300 in `connect_to_inspector_server`.

```c++
bool connect_to_inspector_server()
{
#ifdef __serenity__
    auto socket = Core::LocalSocket::construct();
    if (!socket->connect(SocketAddress::local("/tmp/portal/inspectables")))
        return false;
    s_inspector_server_connection = InspectorServerConnection::construct(move(socket));
    return true;
#else
    VERIFY_NOT_REACHED();
#endif
}
```


As this function was introduced in commit `dc25a4e2`, the tag `regression` on this issue may be misleading.


As for a potential fix, I am not that familiar with Serenity, so I am not sure what would be best. but the 2 fixes I can see are:

1. Unveil  `/tmp/portal/inspectables` in `connect_to_inspector_server()`
or
2. Make each user of `LibCore::EventLoop` that has `make_inspectable == MakeInspectable::Yes` unveil it themselves.

It would be best if someone who actually understands how `LibCore::EventLoop` works would fix this :^)

**Comment by linusg** [comment]
This was fixed in 5f342c41b9d01416aac8998e86054d68699b3e62.
